code/modules/projectiles/guns/beam_rifle.dm

```diff
@@ -472,7 +472,7 @@
 		if(original && (original.layer>=2.75) || ismob(original))
```

Comment:
step_to calls entered, but projectiles aren't always dense and not everything they could be hitting is either, so this ensures bump gets called based on projectile logic even when density doesn't match up.